src/Sets/LineSegment.jl

```diff
@@ -210,7 +210,8 @@ Return the generator matrix of a line segment.
 A matrix with a single column representing the generator of `L`.
 """
 function genmat(L::LineSegment)
-    return genmat_fallback(L)
+    ngens = L.p == L.q ? 0 : 1
```

Comment:
```suggestion
    ngens = _isapprox(L.p, L.q) ? 0 : 1
```

Reply:
The main code checks equality and I just copy the behavior here. I would prefer to add functional changes later.

Reply:
> The main code 

Where is it?

> I would prefer to add functional changes later.

This function introduces a bug, unless we restrict the numeric type of the line segment to an exact ring such as rationals. I agree to defer fixing bugs for later, but do you mind opening a follow-up issue? (if it already exists i didn't notice)

Reply:
#2001